gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -301,18 +296,17 @@ def __init__(self,
               self._process_header_lines),
           **kwargs)
 
-    if isinstance(self, PySamParser):
-      # In an absence of "#CHROM POS..." header line, PySam process would lock.
-      # However, we need to support cases when no records in files were provided
-      # even if the above header line is missing. To circumvent the error, read
-      # the file until the first line is found, and verify validity.
-      self._header_lines = None
-      if isinstance(range_tracker, range_trackers.UnsplittableRangeTracker):
-        secondary_tracker = range_trackers.UnsplittableRangeTracker(
-            copy(range_tracker._range_tracker))
-      else:
-        secondary_tracker = copy(range_tracker)
-      self._verify_header(copy(text_source), secondary_tracker)
+    # In an absence of "#CHROM POS..." header line, PySam process would lock.
+    # However, we need to support cases when no records in files were provided
+    # even if the above header line is missing. To circumvent the error, read
+    # the file until the first line is found, and verify validity.
+    self._header_lines = None
+    if isinstance(range_tracker, range_trackers.UnsplittableRangeTracker):
+      secondary_tracker = range_trackers.UnsplittableRangeTracker(
+          copy(range_tracker._range_tracker))
+    else:
+      secondary_tracker = copy(range_tracker)
+    self._verify_header(copy(text_source), secondary_tracker)
```

Comment:
Not really, since it has to happen before actual read has happened.

However, this method is not really needed, since now vcf_header_io, which always happens before reading actual variants does the same. I was going to do it later, but since it's creating points of contention and especially since it has negative affect on the performance, I've removed it.

Also moved a test to vcf_header_io_test, but it needs expanding to encapsulate more cases. Added a TODO there.